tpm/src/lib.rs

```diff
@@ -0,0 +1,434 @@
+// Copyright © 2022, Microsoft Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use anyhow::anyhow;
+use byteorder::{BigEndian, ReadBytesExt};
+use std::convert::TryInto;
+use thiserror::Error;
+
+pub const TPM_CRB_BUFFER_MAX: usize = 3968; // 0x1_000 - 0x80
+pub const TPM_SUCCESS: u32 = 0x0;
+
+/*
+ * Structures required to process Request and Responses of Control commands
+ * used by control channel over UNIX socket interface
+ *
+ * All messages contain big-endian data.
+ *
+ * Reference: https://github.com/stefanberger/swtpm/blob/master/man/man3/swtpm_ioctls.pod
+ */
+#[derive(Debug, Clone, Copy)]
+pub enum Commands {
+    CmdGetCapability = 1,
+    CmdInit,
+    CmdShutdown,
+    CmdGetTpmEstablished,
+    CmdSetLocality,
+    CmdHashStart,
+    CmdHashData,
+    CmdHashEnd,
+    CmdCancelTpmCmd,
+    CmdStoreVolatile,
+    CmdResetTpmEstablished,
+    CmdGetStateBlob,
+    CmdSetStateBlob,
+    CmdStop,
+    CmdGetConfig,
+    CmdSetDatafd,
+    CmdSetBufferSize,
+}
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("Failed converting buf to PTM : {0}")]
+    ConvertToPtm(#[source] anyhow::Error),
+}
+type Result<T> = anyhow::Result<T, Error>;
+
+#[derive(PartialEq, Eq, Copy, Clone, Debug)]
+pub enum MemberType {
+    Request,
+    Response,
+    Error,
+    Cap,
+}
+
+pub trait Ptm {
+    // Get Member Type
+    fn get_memb_type(&self) -> MemberType;
+
+    // Set Member Type
+    fn set_memb_type(&mut self, mem: MemberType);
+
+    // Convert PTM Request to bytes to be sent to tpm
+    fn ptm_to_request(&self) -> Vec<u8>;
+
+    // Update PTM from tpm's reponse
+    fn update_ptm_with_response(&mut self, buf: &[u8]) -> Result<()>;
+
+    // Update tpm result
+    fn set_res(&mut self, res: u32);
```

Comment:
nit: Shouldn't this be `set_result_code` for consistency?

Reply:
thanks for the catch! updated